synapse/lib/layer.py

```diff
@@ -1530,6 +1530,10 @@ async def _initLayerStorage(self):
         if self.layrvers < 6:
             await self._layrV5toV6()
 
+        if self.layrvers != 6:
```

Comment:
Wouldn't this never be true since the layerV5toV6() could would have updated it?

Reply:
It wouldn't be true if this layer is from the future and this is an accidental downgrade.